build/version.sh

```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/env bash
+
+# Copyright 2020 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit -o nounset -o pipefail
+
+git_dirty=$([[ -z $(git status --short) ]] || echo "-dirty")
+git_tag=$(git tag --points-at HEAD)
+if [ -n "${git_tag}" ]; then
+    # Use the git tag, removing the leading 'v' if it exists.
+    echo "${git_tag/#v/}${git_dirty}"
+else
+    # No git tag found for current commit, the version is unknown, but we use
+    # the short hash for reference.
+    git_hash_short=$(git rev-parse --short HEAD)
+    echo "0.0.0-${git_hash_short}${git_dirty}"
```

Comment:
How about using the last version tag the commit is based on instead of just `0.0.0`? Potentially including the number of commits since this tag, e.g. how the [`quarks-operator`](https://github.com/cloudfoundry-incubator/quarks-operator/blob/master/bin/include/versioning) does it.
That makes it a lot more obvious what you're actually dealing with.

Reply:
I dislike it because it doesn't take into consideration that the latest tag may be a patch version of a previous major release. Then, the version is not always a faithful representation of the code. Honestly, I don't think there's a silver bullet here.

Reply:
Most importantly, the version uses the tag when we want to release it.

Reply:
I think by _last version tag_ @aduffeck means _the closest tag that is an ancestor in the git graph_, and not _the last tag to have been applied to the repository_.  Thus the versioning should always be describing the current commit.

Essentially, the output of `git describe --tags --dirty`.

Reply:
Sounds good, I fixed in https://github.com/kubernetes-sigs/minibroker/pull/84/commits/34382cfb07e73ca1a1fec7169c3ac26cae2a7c78.